### ISSUE TITLE
[bug] Cross compiling with dual profile breaks with fPIC

### ISSUE THREAD

**Issue by anton-danielsson** [conan-io/conan#8443]
I'm unable to use almost any package when cross compiling from Windows to Android.

It seems like the
```py 
   def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC
```
pattern that is used in almost all recipes breaks when I try use the new `--profile:host/--profile:build` logic to cross compile on Windows...

As a work around I tried to replace the `config_options` logic with:
```py
    def package_id(self):
        if self.settings.os == "Windows":
            del self.info.options.fPIC
```
and it seems to work.
It feels like a hack though.

I first create an issue in CCI (https://github.com/conan-io/conan-center-index/issues/4483) but I think it actually belongs here.

### Environment Details (include every applicable attribute)
  * Operating System+version: **Windows 10**
  * Compiler+version: **Visual Studio 16/Android Clang 11**
  * Conan version: **conan 1.33.1**
  * Python version: **Python 3.7.8**

### Conan profile
```
Configuration (profile_host):
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=11
os=Android
os.api_level=22
[options]
[build_requires]
*: android-ndk/r22
[env]

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)

Minimal example:
```py
from conans import ConanFile

class fooConan(ConanFile):
    name = "foo"
    generators = "cmake", "cmake_find_package_multi"

    settings = "os", "arch", "compiler", "build_type"

    requires = (
        "protobuf/3.13.0", # to get Android libprotobuf
    )

    build_requires = (
        "protobuf/3.13.0", # to get Windows proto.exe
    )
```

run:
```bash
conan create . version@dirac/testing \
    --profile:host android-ndk-r20-api-21-arm64-v8a-clang-libcxx.txt \
    --profile:build default \
    --build=outdated
```

You get the error:
```
ERROR: protobuf/3.13.0: option 'fPIC' doesn't exist
Possible options are ['shared', 'with_zlib', 'lite']
```

Having either or of `requires` and `build_requires` works.

### Logs (Include/Attach if Applicable)

<details><summary>Click to expand log</summary>

```
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'url'. It is recommended to add it as attribute
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'license'. It is recommended to add it as attribute
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'description'. It is recommended to add it as attribute
Exporting package recipe
foo/version@dirac/testing: The stored package has not changed
foo/version@dirac/testing: Exported revision: fb9a654636e5e59257bc4d6b6fffaa74
Configuration (profile_host):
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=11
os=Android
os.api_level=22
[options]
[build_requires]
*: android-ndk/r22
[env]

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

android-ndk/r22: Package is up to date
ERROR: protobuf/3.13.0: option 'fPIC' doesn't exist
Possible options are ['shared', 'with_zlib', 'lite']
```

</details>

**Comment by czoido** [comment]
Hi @anton-dirac,
Thanks a lot for reporting, I have been able to reproduce the issue. Looks like the options of the host context are being propagated to the protobuf that is as build_requires in the build context making the build fail. I will mark this as a bug and we will try to understand what's happening there and what can we do to solve it.

**Comment by ytimenkov** [comment]
> host context are being propagated to the protobuf that is as build_requires

Sounds similar to my issue: https://github.com/conan-io/conan-center-index/issues/4013 (and this one, names lockfiles but related to building a graph when same package is in build and ordinary requirement: https://github.com/conan-io/conan/issues/8335)

**Comment by anton-danielsson** [comment]
Yes, looks very similar indeed.

I have also noticed that the values of options are incorrectly passed between build and host requirements.

**Comment by SuTanTank** [comment]
I'm running into similar problem when trying to build protobuf windows -> android. 
I'm using the recipe from conancenter. [https://github.com/conan-io/conan-center-index/blob/master/recipes/protobuf/all/conanfile.py](url)
I use the profile from this thread by @anton-danielsson 
```
Exporting package recipe
protobuf/3.17.1 exports: File 'conandata.yml' found. Exporting it...
protobuf/3.17.1 exports: Copied 1 '.yml' file: conandata.yml
protobuf/3.17.1 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
protobuf/3.17.1 exports_sources: Copied 4 '.patch' files: upstream-issue-7567-no-export-template-define.patch, upstream-pr-7288-android-log.patch, upstream-pr-7761-cmake-regex-fix.patch, upstream-pr-8301-bundle-destination.patch
protobuf/3.17.1: A new conanfile.py version was exported
protobuf/3.17.1: Folder: C:\Users\Tan\.conan\data\protobuf\3.17.1\_\_\export
protobuf/3.17.1: Exported revision: 7c958aa031242f807358a039372a100d
Configuration (profile_host):
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=c++_static
compiler.version=11
os=Android
os.api_level=21
[options]
[build_requires]
*: android-ndk/r22
[env]

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

protobuf/3.17.1: Forced build from source
ERROR: protobuf/3.17.1: option 'fPIC' doesn't exist
Possible options are ['shared', 'with_zlib', 'with_rtti', 'lite']
```

I can't find where the error is raised, seems it's inside Conan code.

**Comment by hpesoj** [comment]
I tried the quoted workaround, but it does nothing:

```
    def package_id(self):
        if self.settings.os == "Windows":
            del self.info.options.fPIC
```

Is there a workaround for this? It's kind of a showstopper if not.

**Comment by SuTanTank** [comment]
I ended up packaging prebuilt binaries for these 3rd party libs, which are already available in our team and the build scripts are actually working.

**Comment by SeanSnyders** [comment]
Also hitting this problem when cross-compiling GTest to Android on Windows.
![image](https://user-images.githubusercontent.com/1384135/141366069-ac6b5a98-219e-4d5d-bdff-de0bc4c03600.png)

When I do the same on macOS, it works (obviously the profiles for the build and host differ between Windows and macOS).

On Windows I use this resolved profile for the host (i.e Android):
```
[settings]
os=Android
os_build=Windows
arch=armv8
arch_build=x86_64
compiler=clang
compiler.version=11
build_type=Release
os.api_level=21 # minimum android version supported
compiler.libcxx=libc++
[options]
[conf]
[build_requires]
*: cmake/3.19.4, android-ndk/r22, mingw_installer/1.0@conan/stable
[env]
```

and build profile:
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=gcc
build_type=Release
compiler.version=6
compiler.libcxx=libstdc++11
compiler.exception=seh
compiler.threads=posix
[options]
[conf]
[build_requires]
*: cmake/3.19.4, mingw_installer/1.0@conan/stable
[env]
```

On Windows we require the `make` program to be set. The Android profile for the host platform adds this dynamically via a MinGW build requirement if it detects the Windows platform and it uses `clang` from the included Android NDK build requirement, but for the build platform we thus specify to use MinGW with `gcc`.

**Comment by SeanSnyders** [comment]
I can also confirm that the quoted workaround
```
def package_id(self):
        if self.settings.os == "Windows":
            del self.info.options.fPIC
```
does not work.

**Comment by Jonkrist** [comment]
We also encountered this recently.
It's been a year since it was first reported.
Any prospect of a solution?

**Comment by Mikayex** [comment]
Also having the same problem, but with shaderc package...

**Comment by zhizhi-dev** [comment]
Mark . waiting for to be done on this problem.

**Comment by xandox** [comment]
Hello. Any news here? Run in the same issue with flatbuffers.

**Comment by memsharded** [comment]
Sorry about this being there for so long. I'll have a look, it was labeled as complex:high back then, lets see...

**Comment by memsharded** [comment]
I have been able to reproduce the error, with this test: https://github.com/conan-io/conan/pull/12904 (this fails if used in 1.X branch)

The good news is that we were kind of aware of this, and we took it into account while preparing the 2.0, so it is already solved there.
The bad news is that it is still very complex to fix in 1.X. I am proposing https://github.com/conan-io/conan/pull/12904 to close this ticket for 2.0 scope (due next month), and I will try to find a workaround for 1.X, but seems challenging.

**Comment by memsharded** [comment]
We have decided to move forward with https://github.com/conan-io/conan/pull/12905, and it has been merged for 1.58.
This will unblock the issue, conditioned to using ``build`` profile, as that is a feature "under development", so hopefully this will not be such a big risk. Thanks everyone for the feedback.
